runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/block/CrailFileBlock.java

```diff
@@ -0,0 +1,401 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.nemo.runtime.executor.data.block;
+
+import org.apache.crail.*;
+import org.apache.nemo.common.KeyRange;
+import org.apache.nemo.common.Pair;
+import org.apache.nemo.common.exception.BlockFetchException;
+import org.apache.nemo.common.exception.BlockWriteException;
+import org.apache.nemo.runtime.executor.data.DataUtil;
+import org.apache.nemo.runtime.executor.data.FileArea;
+import org.apache.nemo.runtime.executor.data.metadata.FileMetadata;
+import org.apache.nemo.runtime.executor.data.metadata.PartitionMetadata;
+import org.apache.nemo.runtime.executor.data.partition.NonSerializedPartition;
+import org.apache.nemo.runtime.executor.data.partition.Partition;
+import org.apache.nemo.runtime.executor.data.partition.SerializedPartition;
+import org.apache.nemo.runtime.executor.data.streamchainer.Serializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.io.*;
+import java.util.*;
+
+/**
+ * This class represents a block which is stored in CrailStore.
+ * Concurrent read is supported, but concurrent write is not supported.
+ *
+ * @param <K> the key type of its partitions.
+ */
+@NotThreadSafe
+public final class CrailFileBlock<K extends Serializable> implements Block<K> {
+  private static final Logger LOG = LoggerFactory.getLogger(CrailFileBlock.class.getName());
+  private final String id;
+  private final Map<K, SerializedPartition<K>> nonCommittedPartitionsMap;
+  private final Serializer serializer;
+  private final String filePath;
+  private final FileMetadata<K> metadata;
+  private CrailStore fs = null;
+  private CrailFile file = null;
+
+  /**
+   * Constructor.
+   *
+   * @param blockId    the ID of this block.
+   * @param serializer the {@link Serializer}.
+   * @param filePath   the path of the file that this block will be stored.
+   * @param metadata   the metadata for this block.
+   * @param fs         CrailStore instance of the Crail storage.
+   */
+
+  public CrailFileBlock(final String blockId,
+                        final Serializer serializer,
+                        final String filePath,
+                        final FileMetadata<K> metadata,
+                        final CrailStore fs) {
+    this.id = blockId;
+    this.nonCommittedPartitionsMap = new HashMap<>();
+    this.serializer = serializer;
+    this.filePath = filePath;
+    this.metadata = metadata;
+      try {
```

Comment:
remove two empty spaces before `try`. Please check throughout the code. 

If you use Vim:
https://stackoverflow.com/questions/506075/how-do-i-fix-the-indentation-of-an-entire-file-in-vi